-------------------------------------------------------------------------------
Test set: TestProjet$calculTotalProjetAchat
-------------------------------------------------------------------------------
Tests run: 5, Failures: 2, Errors: 2, Skipped: 0, Time elapsed: 0.080 s <<< FAILURE! -- in TestProjet$calculTotalProjetAchat
TestProjet$calculTotalProjetAchat.calculTotalProjetAchatArrondi -- Time elapsed: 0.003 s
TestProjet$calculTotalProjetAchat.calculTotalProjetAchatSimple -- Time elapsed: 0.054 s <<< FAILURE!
org.opentest4j.MultipleFailuresError: 
Multiple Failures (2 failures)
	org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 
Integer cannot be returned by calculDroitEnregistrement()
calculDroitEnregistrement() should return double
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <-2399.9999463558197>
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:80)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:44)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:38)
	at org.junit.jupiter.api.Assertions.assertAll(Assertions.java:2940)
	at TestProjet$calculTotalProjetAchat.calculTotalProjetAchatSimple(TestProjet.java:128)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	Suppressed: org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 
Integer cannot be returned by calculDroitEnregistrement()
calculDroitEnregistrement() should return double
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

		at be.iramps.ue1103.Pret.Projet.calculDroitEnregistrement(Projet.java:54)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatSimple$0(TestProjet.java:134)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <-2399.9999463558197>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatSimple$4(TestProjet.java:163)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more

TestProjet$calculTotalProjetAchat.calculTotalProjetAchatSimpleAvecMockito -- Time elapsed: 0.002 s <<< ERROR!
org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 

Integer cannot be returned by calculDroitEnregistrement()
calculDroitEnregistrement() should return double
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	at be.iramps.ue1103.Pret.Projet.calculDroitEnregistrement(Projet.java:54)
	at TestProjet$calculTotalProjetAchat.calculTotalProjetAchatSimpleAvecMockito(TestProjet.java:170)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

TestProjet$calculTotalProjetAchat.calculTotalProjetAchatArrondiAvecMockito -- Time elapsed: 0.002 s <<< ERROR!
org.mockito.exceptions.misusing.WrongTypeOfReturnValue: 

Integer cannot be returned by calculDroitEnregistrement()
calculDroitEnregistrement() should return double
***
If you're unsure why you're getting above error read on.
Due to the nature of the syntax above problem might occur because:
1. This exception *might* occur in wrongly written multi-threaded tests.
   Please refer to Mockito FAQ on limitations of concurrency testing.
2. A spy is stubbed using when(spy.foo()).then() syntax. It is safer to stub spies - 
   - with doReturn|Throw() family of methods. More in javadocs for Mockito.spy() method.

	at be.iramps.ue1103.Pret.Projet.calculDroitEnregistrement(Projet.java:54)
	at TestProjet$calculTotalProjetAchat.calculTotalProjetAchatArrondiAvecMockito(TestProjet.java:244)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

TestProjet$calculTotalProjetAchat.calculTotalProjetAchatNegatif -- Time elapsed: 0.009 s <<< FAILURE!
org.opentest4j.MultipleFailuresError: 
Multiple Failures (5 failures)
	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <163359.2>
	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <-113656.19982576638>
	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <31306.61999514252>
	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <45407.795>
	org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <181724.27986622616>
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:80)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:44)
	at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:38)
	at org.junit.jupiter.api.Assertions.assertAll(Assertions.java:2940)
	at TestProjet$calculTotalProjetAchat.calculTotalProjetAchatNegatif(TestProjet.java:184)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <163359.2>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatNegatif$5(TestProjet.java:190)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <-113656.19982576638>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatNegatif$6(TestProjet.java:197)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <31306.61999514252>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatNegatif$7(TestProjet.java:204)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <45407.795>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatNegatif$8(TestProjet.java:211)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more
	Suppressed: org.opentest4j.AssertionFailedError: expected: <java.lang.Exception> but was: <181724.27986622616>
		at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
		at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
		at org.junit.jupiter.api.AssertEquals.failNotEqual(AssertEquals.java:197)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:182)
		at org.junit.jupiter.api.AssertEquals.assertEquals(AssertEquals.java:177)
		at org.junit.jupiter.api.Assertions.assertEquals(Assertions.java:1141)
		at TestProjet$calculTotalProjetAchat.lambda$calculTotalProjetAchatNegatif$9(TestProjet.java:218)
		at org.junit.jupiter.api.AssertAll.lambda$assertAll$0(AssertAll.java:68)
		at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:215)
		at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
		at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:570)
		at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:560)
		at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
		at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
		at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:727)
		at org.junit.jupiter.api.AssertAll.assertAll(AssertAll.java:77)
		... 8 more

